omniscidb/StringDictionary/StringDictionaryProxy.cpp

```diff
@@ -42,7 +42,13 @@ constexpr int32_t transient_id_ceil{-2};
 
 StringDictionaryProxy::StringDictionaryProxy(std::shared_ptr<StringDictionary> sd,
                                              const int64_t generation)
-    : string_dict_(sd), generation_(generation) {}
+    : string_dict_(sd), generation_(generation) {
+  if (generation_ < 0) {
+    generation_ = static_cast<int64_t>(sd->storageEntryCount());
```

Comment:
Do you plan to forbid negative values later on?

Reply:
Yes, it will be in the next patch.